### PR TITLE
zls: 0.11.0 -> 0.12.0

### PR DESCRIPTION
**Pull request by einetuer**
## Description of changes
Diff: https://github.com/zigtools/zls/compare/0.11.0...0.12.0

Changelog: https://github.com/zigtools/zls/releases/tag/0.12.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/language-servers/zls/default.nix [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
     owner = "zigtools";
     repo = "zls";
     rev = finalAttrs.version;
     fetchSubmodules = true;
-    hash = "sha256-WrbjJyc4pj7R4qExdzd0DOQ9Tz3TFensAfHdecBA8UI=";
+    hash = "sha256-2iVDPUj9ExgTooDQmCCtZs3wxBe2be9xjzAk9HedPNY=";
   };
 
+  langref = fetchurl {
+    url = "https://raw.githubusercontent.com/ziglang/zig/a685ab1499d6560c523f0dbce2890dc140671e43/doc/langref.html.in";
```
Comment (AndersonTorres) [inline pkgs/development/tools/language-servers/zls/default.nix:22]: Can this file be extracted from `zig.src`?

**Review by iFreilicht — APPROVED** [review]
Mostly trivial update. Built successfully locally on x86_64-linux and aarch64-darwin with nixpkgs-review. Built binary launched successfully and has the expected version.

Merging will unblock https://github.com/NixOS/nixpkgs/pull/306077, which is part of the 24.05 milestone.

File: pkgs/development/tools/language-servers/zls/default.nix (hunk 1 above)
Comment (0x5a4) [reply pkgs/development/tools/language-servers/zls/default.nix:22]: It is part of the source tree, so propably yes. I'll try tomorrow

**Review by moni-dz — APPROVED** [review]

**Comment by 0x5a4** [comment]
As @nbraud mentioned, zls 0.12 isnt backward compatible with zig 0.11. Should we split this into zls_0_11 and zls_0_12 then?

**Review by nicoonoclaste — APPROVED** [review]

**Pull request merged**